docs/explanation/security/cryptographic-approaches.rst

```diff
@@ -0,0 +1,144 @@
+  The store implements both automated checks and manual reviews to ensure the quality and security of the software it hosts.
+
+Resources used by MicroCeph snaps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interfaces
+^^^^^^^^^^
+
+MicroCeph snaps can use the following snap interfaces:
+
+* ``kernel-module-load``: to enable loading of Ceph-specific kernel modules, such as the ``rbd`` module.  
+* ``content``: to expose Ceph configuration to cooperating snaps.
+
+Plugs
+^^^^^
+
+MicroCeph snaps can use the following snap plugs:
+
+* ``block-devices``: for storage  
+* ``dm-crypt``: for disk encryption  
+* ``hardware-observe``: block device detection  
+* ``home``: access to user-supplied configuration and certificates  
+* ``microceph-support``: for additional storage  
+* ``mount-observe``: storage management  
+* ``network``: networking client  
+* ``network-bind``: network servers
+
+Microcluster security
+---------------------
+
+MicroCeph is built on the Canonical Microcluster library. Microcluster manages cluster topology and cluster membership.
+Adding nodes is regulated by issuing tokens which new nodes can use to trigger a request to join the cluster.
+After successfully joining the cluster, Microcluster issues TLS certificates to new nodes. ``TLS1.3``, by default, is the minimum
+supported version. This can be overridden by setting an environment variable to set the minimum supported version to ``TLS1.2``.
+
+Cluster members use the newly created certificates to authenticate for cluster API access.
+
+Ceph authentication and authorization
+-------------------------------------
+
+Internally, MicroCeph deploys a Ceph cluster which uses the ``cephx`` protocol for authentication and
+authorization. See `Cryptography in Ubuntu Ceph
+<https://ubuntu.com/ceph/docs/cryptographic-technologies-in-charmed-ceph#p-151613-cryptography-in-ubuntu-ceph>`_ for more details.
+
+Data at rest
+------------
+
+MicroCeph offers full disk encryption (FDE) for Object Storage Daemons (OSDs), activated when adding disks. Note that this disk encryption
+only pertains to user data managed in Ceph, and not encryption of the cluster data (such as administrative data, 
+logs, etc.).
+To use FDE, users must first connect the ``dm-crypt`` plug for MicroCeph (it is not auto-connected).
+When FDE is requested, MicroCeph generates a random key and stores it in the Ceph Monitor (MON). This key is then used to setup
+Linux Unified Key Setup (LUKS) via ``cryptsetup``, using ``cipher AES-XTS-plain64`` and ``SHA256`` hashing, with a 256-bit keysize.
+
+.. note::
+    While the FDE approach for OSD encryption shares some of the techniques employed by the data at rest
+    encryption features in Ubuntu Ceph, it's a separate implementation due to the specific sandboxing needs of the MicroCeph snap.
+
+Storage types
+-------------
+
+Like Ceph, MicroCeph provides three types of storage to clients via specific components, such as:
```

Comment:
Mhm unsure about the wording here, now that I look at it again :-) 
The subsections below describe types of storage not "components", maybe could be improved with:

> MicroCeph provides the same three types of storage to clients as Ceph. Each type of storage supports specific security features:

Reply:
The subsections are not all types of storage, i.e. dashboard.
What do you think of this wording?:

```
Like Ceph, MicroCeph provides three types of storage, i.e. object, block and file storage, to clients. It does so
via specific components that support specific security features.
```